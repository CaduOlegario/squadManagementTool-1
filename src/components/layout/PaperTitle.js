import {withStyles} from "@material-ui/core";
import Typography from "@material-ui/core/Typography";
import React from "react";
import PropTypes from "prop-types";

const styles = {
    container: {
        padding: "28px",
        borderBottom: "1px solid #ececec",
    }
};

const PaperTitle = (props) => {
    const {classes, title} = props;

    return (
        <div className={classes.container}>
            <Typography variant="h1"> {title} </Typography>
        </div>
    );
};

PaperTitle.propTypes = {
    title: PropTypes.string.isRequired,
    classes: PropTypes.object.isRequired
};

export default withStyles(styles)(PaperTitle);