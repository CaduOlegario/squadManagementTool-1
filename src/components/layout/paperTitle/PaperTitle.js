import {withStyles} from "@material-ui/core";
import Typography from "@material-ui/core/Typography";
import React from "react";
import PropTypes from "prop-types";

const styles = {
    container: {
        padding: "28px",
        borderBottom: "1px solid #ececec",
    }
};

/** Title component for paper grid items with a bottom border*/
const PaperTitle = (props) => {
    const {classes, title} = props;

    return (
        <div className={classes.container}>
            <Typography variant="h1"> {title} </Typography>
        </div>
    );
};

PaperTitle.propTypes = {
    /** Text of the title */
    title: PropTypes.string.isRequired,
    /** @ignore */
    classes: PropTypes.object.isRequired
};

export default withStyles(styles)(PaperTitle);