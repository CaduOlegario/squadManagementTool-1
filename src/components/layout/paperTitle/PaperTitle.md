Paper title example:

```js padded
import Paper from "@material-ui/core/Paper/Paper";

<Paper elevation={1} style={{height: "100%", borderRadius: "14px"}}>
    <PaperTitle
        title="Card Title"
     />
    <div style={{width: "100%", textAlign: "center", padding: "20px"}}>
        additional content
    </div>
</Paper>
```

