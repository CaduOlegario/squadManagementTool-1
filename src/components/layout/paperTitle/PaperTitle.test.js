import React from 'react';
import renderer from 'react-test-renderer';
import PaperTitle from "./PaperTitle";

it('renders correctly normal input', () => {
    const tree = renderer
        .create(<PaperTitle title="Test title"/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
