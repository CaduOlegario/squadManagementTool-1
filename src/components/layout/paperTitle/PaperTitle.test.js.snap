// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly normal input 1`] = `
<div
  className="PaperTitle-container-1"
>
  <h1
    className="MuiTypography-root MuiTypography-h1"
  >
     
    Test title
     
  </h1>
</div>
`;
